### PR TITLE
chore(checkbox): update styles to match library

### PR DESCRIPTION
**Pull request by jinlee93**
### Summary:
[sc-202569]
- Icons are moving towards sharp icons, this updates the checkbox svgs to be sharp
- The previous checkbox icon was a custom svg in-file element and this uses the checkbox icons from the library to follow our icon practices
- Cleans up the checkbox styles and the `--checkbox-svg-size` css custom variable we used to align the label with the checkbox. Uses a flexbox centering strategy instead and is hence also more readable (imo)
- Potentially breaking: removes a small amount of space under the checkbox that used to exist 
image
### Test Plan:
- wrote unit test and updated snaps
- visual regressions only where checkboxes used (Filters, Fieldset, Checkboxes)

### CODE REVIEW

File: src/components/Checkbox/Checkbox.stories.tsx [hunk 1]
```diff
@@ -98,14 +99,13 @@ export const Disabled = {
...
 export const WithoutVisibleLabel: StoryObj<Args> = {
-  ...Default,
   args: {
     'aria-label': 'a checkbox has no name',
     label: undefined,
   },
   render: (args) => (
     <>
-      <Checkbox {...args} />
+      <Checkbox {...args} readOnly />
```
Comment (jinlee93) [inline src/components/Checkbox/Checkbox.stories.tsx:108]: Stories now (correctly) throw error if no `readOnly` or `onChange`

**Comment by dierat** [comment]
Is it expected that the checkbox is smaller than before?

File: src/components/Checkbox/Checkbox.module.css [hunk 2]
```diff
@@ -1,20 +1,13 @@
...
- *    a flexbox parent
  */
 .checkbox {
-  /* 1 */
-  position: relative;
-
-  display: grid; /* 2 */
-  align-items: flex-start;
-  grid-template-columns: min-content 1fr;
+  display: flex;
```
Comment (dierat) [inline src/components/Checkbox/Checkbox.module.css:11]: nice

**Comment by dierat** [comment]
Could you make an alpha build and test this PR in LP before merging? They seem a bit fragile in terms of styling.

File: src/components/Checkbox/Checkbox.test.tsx [hunk 3]
```diff
@@ -18,4 +22,12 @@ describe('<Checkbox />', () => {
     // eslint-disable-next-line testing-library/no-node-access
     expect(container.firstChild).toMatchSnapshot();
   });
+
+  test('should toggle the checkbox with space', () => {
```
Comment (dierat) [inline src/components/Checkbox/Checkbox.test.tsx:26]: Nice!

**Comment by dierat** [comment]
The custom positioning story looks off now btw.

Screen Shot 2022-08-12 at 7 06 54 AM

File: src/components/Checkbox/Checkbox.module.css (hunk 2 above)
Comment (dierat) [reply src/components/Checkbox/Checkbox.module.css:11]: flex for president

File: src/components/CheckboxInput/CheckboxInput.module.css [hunk 4]
```diff
@@ -5,69 +5,67 @@
...
-   *    especially on touch screen readers, still interact with the real checkbox element
-   *    where it would naturally be present.
-   */
+ * The visually hidden input element for the checkbox. The visual checkmark is provided by an svg element.
+ * 1) Removes default margins placed by browser for checkboxes.
+ * 2) Remove the checkbox from the page flow, positioning it on top of the SVG.
+ * 3) Hide the input element visually.
+ *    This ensures the checkbox is still "physically" present so that all users,
+ *    especially on touch screen readers, still interact with the real checkbox element
+ *    where it would naturally be present.
```
Comment (dierat) [inline src/components/CheckboxInput/CheckboxInput.module.css:27]: #3 and 4 have the same text. Is this supposed to be
```suggestion
 * 3) Set same dimensions as the CheckboxSvg element.
```

Comment (jinlee93) [reply src/components/CheckboxInput/CheckboxInput.module.css:27]: Yesssss thanks for the catch 👍

**Comment by jinlee93** [comment]
> The custom positioning story looks off now btw.
Nice catch, fixed

**Comment by jinlee93** [comment]
> Is it expected that the checkbox is smaller than before in storybook?
- Yes, the previous checkbox was 20px and the new one is 18px [mocks](https://www.figma.com/file/Ff7hJoSvXX9hkAmIklnsT2/EDS-Paper-v1.0-%7C-UI-Components?node-id=1591%3A13116)

> Could you make an alpha build and test this PR in LP before merging? They seem a bit fragile in terms of styling.
- Yes great call, will do

**Comment by jinlee93** [comment]
> Could you make an alpha build and test this PR in LP before merging? They seem a bit fragile in terms of styling.

@dierat 
Made an alpha build `3.1.1-alpha.2` for testing in LP. Here is the chromatic [build](https://www.chromatic.com/build?appId=614b969675171d003ac6f98a&number=11440) for [branch](https://github.com/FB-PLP/traject/tree/jlee/testCheckboxStylesUpdates) `jlee/testCheckboxStylesUpdates` where 3.1.1-alpha.2 is used. [Stories](https://614b969675171d003ac6f98a-tehcwwhqdn.chromatic.com/) look good as far as I can tell except for [Multiselect](https://614b969675171d003ac6f98a-tehcwwhqdn.chromatic.com/?path=/story/core-edscandidates-multiselect--default) where clicking the dropdown option does not toggle the checkbox unless it's clicked specifically in the checkbox icon or the label. I think this is behavior is fine as long as the cursor is not `pointer` for the area where it toggle the checkbox.

I'll change the cursor behavior in that branch and land once these checkbox styles changes have been released

**Review by dierat — APPROVED** [review]

**Pull request merged**